### PR TITLE
[release] add 7.21.0-rc.1 build entry.

### PR DESCRIPTION
**Pull request by remeh**
### What does this PR do?

Add 7.21.0-rc.1 build entries.

### Motivation

Build 7.21.0 first release candidate.

### CODE REVIEW

File: release.json [hunk 1]
```diff
@@ -13,6 +13,20 @@
         "JMXFETCH_VERSION": "0.38.0",
         "JMXFETCH_HASH": "f21fe4d604d56adb74423b940892178677cba635d6f1429c1606fc753c99aaed"
     },
+    "7.21.0-rc.1": {
+        "INTEGRATIONS_CORE_VERSION": "7.21.0-rc.1",
+        "OMNIBUS_SOFTWARE_VERSION": "7.21.0-rc.1",
+        "OMNIBUS_RUBY_VERSION": "datadog-5.5.0",
```
Comment (mx-psi) [inline release.json:19]: Why did the omnibus version scheme change from 7.20?

Comment (remeh) [reply release.json:19]: Nice catch 🙇

Comment (remeh) [reply release.json:19]: (took nightly entries)

**Review by mx-psi — APPROVED** [review]
LGTM once pending PRs and integration tag are addressed

**Review by FlorianVeaux — APPROVED** [review]
Integration tag has been pushed to the repository: https://github.com/DataDog/integrations-core/tree/7.21.0-rc.1

Looks good for the integrations team

**Pull request merged**